bootstrap/eks/api/v1beta1/conversion.go

```diff
@@ -93,3 +93,8 @@ func (r *EKSConfigTemplateList) ConvertFrom(srcRaw conversion.Hub) error {
 
 	return Convert_v1beta2_EKSConfigTemplateList_To_v1beta1_EKSConfigTemplateList(src, r, nil)
 }
+
```

Comment:
I can't comment on the line above, but what's missing here is to convert these values:
```
// WARNING: in.PreBootstrapCommands requires manual conversion: does not exist in peer-type
	// WARNING: in.PostBootstrapCommands requires manual conversion: does not exist in peer-type
	// WARNING: in.BootstrapCommandOverride requires manual conversion: does not exist in peer-type
	// WARNING: in.Files requires manual conversion: does not exist in peer-type
	// WARNING: in.DiskSetup requires manual conversion: does not exist in peer-type
	// WARNING: in.Mounts requires manual conversion: does not exist in peer-type
	// WARNING: in.Users requires manual conversion: does not exist in peer-type
	// WARNING: in.NTP requires manual conversion: does not exist in peer-type
```

These should be done manually in the above ConvertTo and ConvertFrom methods.

For example, like here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3741/files#diff-a86bb4e63de0ea453a7d13e98a2d9b8b7de1181d7bfa542a67e76e7d711678baR36

Reply:
Thanks for providing an example, I was able to fill in the ConvertTo and ConvertFrom and restore the conversion_test that I deleted to make the tests pass earlier.

Are there any additional docs on this conversion behavior? I still feel like there's a bit happening under the covers in the conversion that I don't follow.

Reply:
Basically, they are converted to annotations and then those annotations are restored to actual values. :)

Yeah, the conversion Fuzzy test makes sure that the conversion ( when people upgrade for example ) work in these scenarios.